npms/skale-ima/index.js

```diff
@@ -2502,6 +2506,31 @@ async function do_transfer(
                     );
                 }
                 //
+                //
+                //
+                { // EXPERIMENTAL BLOCK: check DST permissions, result is ignored
+                    if( verbose_get() >= RV_VERBOSE.trace )
+                        log.write( strLogPrefix + cc.debug( "Will check permissions in destination message proxy..." ) + "\n" );
+                    try {
+                        const a = joAccountDst.address( w3_dst );
+                        const hashOfSchainName = w3_dst.utils.keccak256( chain_id_src );
+                        if( verbose_get() >= RV_VERBOSE.trace ) {
+                            log.write( strLogPrefix + cc.debug( "From address " ) + cc.info( a ) + "\n" );
+                            log.write( strLogPrefix + cc.debug( "Chain name " ) + cc.info( chain_id_src ) + "\n" );
+                            log.write( strLogPrefix + cc.debug( "Hash of chain name " ) + cc.info( hashOfSchainName ) + "\n" );
+                        }
+                        const r = await jo_message_proxy_dst.methods.isAuthorizedCaller( hashOfSchainName, a ).call( {
```

Comment:
This will work only if DST is "Mainnet". MessageProxy on Schain does not have this call. Maybe let's rename checkcheckIsAuthorizedCaller to isAuthorizedCaller in MessageProxy on Schain?

Reply:
yes, I had to put it into try/catch block to deal with not-existing contract method

Reply:
renamed `checkcheckIsAuthorizedCaller` to `isAuthorizedCaller`